packages/app/src/cli/services/dev/app-events/file-watcher.ts

```diff
@@ -91,116 +127,83 @@ export async function startFileWatcher(
         if (!isToml) {
-          pushEvent({type: 'file_created', path, extensionPath, startTime})
+          this.pushEvent({type: 'file_created', path, extensionPath, startTime})
           break
         }
         let totalWaitedTime = 0
         const realPath = dirname(path)
         const intervalId = setInterval(() => {
           if (fileExistsSync(joinPath(realPath, configurationFileNames.lockFile))) {
             outputDebug(`Waiting for extension to complete creation: ${path}\n`)
-            totalWaitedTime += 500
+            totalWaitedTime += EXTENSION_CREATION_CHECK_INTERVAL_IN_MS
           } else {
             clearInterval(intervalId)
-            extensionPaths.push(realPath)
-            pushEvent({type: 'extension_folder_created', path: realPath, extensionPath, startTime})
+            this.extensionPaths.push(realPath)
+            this.pushEvent({type: 'extension_folder_created', path: realPath, extensionPath, startTime})
           }
-          if (totalWaitedTime >= 20000) {
+          if (totalWaitedTime >= EXTENSION_CREATION_TIMEOUT_IN_MS) {
             clearInterval(intervalId)
-            options.stderr.write(`Extension creation detection timeout at path: ${path}\nYou might need to restart dev`)
+            this.options.stderr.write(
+              `Extension creation detection timeout at path: ${path}\nYou might need to restart dev`,
+            )
           }
-        }, 200)
+        }, EXTENSION_CREATION_CHECK_INTERVAL_IN_MS)
         break
       case 'unlink':
         // Ignore shoplock files
         if (path.endsWith(configurationFileNames.lockFile)) break
 
         if (isConfigAppPath) {
-          pushEvent({type: 'app_config_deleted', path, extensionPath, startTime})
+          this.pushEvent({type: 'app_config_deleted', path, extensionPath, startTime})
         } else if (isToml) {
           // When a toml is deleted, we can consider every extension in that folder was deleted.
-          extensionPaths = extensionPaths.filter((extPath) => extPath !== extensionPath)
-          pushEvent({type: 'extension_folder_deleted', path: extensionPath, extensionPath, startTime})
+          this.extensionPaths = this.extensionPaths.filter((extPath) => extPath !== extensionPath)
+          this.pushEvent({type: 'extension_folder_deleted', path: extensionPath, extensionPath, startTime})
         } else {
-          // This could be an extension delete event, Wait 500ms to see if the toml is deleted or not.
           setTimeout(() => {
             // If the extensionPath is not longer in the list, the extension was deleted while the timeout was running.
-            if (!extensionPaths.includes(extensionPath)) return
-            pushEvent({type: 'file_deleted', path, extensionPath, startTime})
+            if (!this.extensionPaths.includes(extensionPath)) return
+            this.pushEvent({type: 'file_deleted', path, extensionPath, startTime})
           }, 500)
```

Comment:
Should we change this `500` to a timeout const too, like we're doing with the rest of the times in this file?

Reply:
yep, good catch, will change it